62.CAD/fragment_shader.hlsl

```diff
@@ -256,13 +253,46 @@ float4 main(PSInput input) : SV_TARGET
         const float antiAliasingFactor = globals.antiAliasingFactor;
         localAlpha = 1.0f - smoothstep(-antiAliasingFactor, +antiAliasingFactor, distance);
     }
-    /*
-    TODO[Lucas]:
-        Another else case for CurveBox where you simply do what I said in the notes of common.hlsl PSInput
-        and solve two quadratic equations, you could check for it being a "line" for the mid point being nan
-        you will use input.getXXX() to get values needed for this computation
-    */
-    
+    else if (objType == ObjectType::CURVE_BOX) 
+    {
+        const float minorBBoxUv = input.getMinorBBoxUv();
+        const float majorBBoxUv = input.getMajorBBoxUv();
+
+        nbl::hlsl::equations::Quadratic<float> curveMinMinor = input.getCurveMinMinor();
+        nbl::hlsl::equations::Quadratic<float> curveMinMajor = input.getCurveMinMajor();
+        nbl::hlsl::equations::Quadratic<float> curveMaxMinor = input.getCurveMaxMinor();
+        nbl::hlsl::equations::Quadratic<float> curveMaxMajor = input.getCurveMaxMajor();
+
+        const float minT = clamp(input.getMinCurvePrecomputedRootFinders().computeRoots().x, 0.0, 1.0);
+        const float minEv = curveMinMinor.evaluate(minT);
+        
+        const float maxT = clamp(input.getMaxCurvePrecomputedRootFinders().computeRoots().x, 0.0, 1.0);
+        const float maxEv = curveMaxMinor.evaluate(maxT);
+        
+        float2 tangentMinCurve = float2(
+            2.0 * curveMinMinor.a * minT + curveMinMinor.b,
+            2.0 * curveMinMajor.a * minT + curveMinMajor.b);
+        tangentMinCurve = normalize(tangentMinCurve / float2(fwidth(minorBBoxUv), fwidth(majorBBoxUv)));
```

Comment:
@devshgraphicsprogramming suggested `length(ddx, ddy)` instead of `fwidth`, so let's do that